### PR TITLE
add usage examples

### PR DESCRIPTION
**Pull request by WilliamHYZhang**
added usage examples for tf.image.rgb_to_grayscale and tf.image.grayscale_to_rgb

### CODE REVIEW

File: tensorflow/python/ops/image_ops_impl.py [hunk 1]
```diff
@@ -1874,6 +1874,13 @@ def rgb_to_grayscale(images, name=None):
 
   Returns:
     The converted grayscale image(s).
+    
+  Usage Example:
+   ```python
+   import tensorflow as tf
```
Comment (alextp) [inline tensorflow/python/ops/image_ops_impl.py]: No need for importing tensorflow

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]
Please indent the code blocks in the documentation the same way in both blocks

File: tensorflow/python/ops/image_ops_impl.py [hunk 2]
```diff
@@ -1874,6 +1874,13 @@ def rgb_to_grayscale(images, name=None):
...
+   original = tf.random.normal(shape=(256, 256, 3))
+   converted = tf.image.rgb_to_grayscale(original)
```
Comment (alextp) [inline tensorflow/python/ops/image_ops_impl.py]: Can you choose a sufficiently small image with nice floating point values (a 1x1 image would work here, for example) so we can make this a doctest?

**Review by alextp — CHANGES_REQUESTED** [review]

File: tensorflow/python/ops/image_ops_impl.py (hunk 2 above)
Comment (WilliamHYZhang) [reply tensorflow/python/ops/image_ops_impl.py]: Ok, shall I replace line 1881 with `original = tf.random.normal(shape=(1, 1, 3))` instead? Default output value should already be tf.dtypes.float32, would that need to change as well? Thanks for the feedback.

File: tensorflow/python/ops/image_ops_impl.py (hunk 1 above)
Comment (WilliamHYZhang) [reply tensorflow/python/ops/image_ops_impl.py]: Ok, I will get rid of that. Thanks for the feedback.

**Comment by WilliamHYZhang** [comment]
@alextp @mihaimaruseac I made the requested changes. Thanks!

**Comment by mihaimaruseac** [comment]
I think @alextp also wanted you to print the value of `converted`, basically to make this a [Python doctest](https://docs.python.org/2/library/doctest.html). See for example https://github.com/tensorflow/tensorflow/blob/f70c46f8cd2a91b390455827cda65b5b5fe92ef0/tensorflow/python/framework/ops.py#L446-L468

**Review by alextp — CHANGES_REQUESTED** [review]
Now can you add >>> to make these doctests? And replace the random input with a constant array to make the test deterministic?

**Comment by WilliamHYZhang** [comment]
Would you like me to also add `print(converted)` as @mihaimaruseac suggested? When doing that, however, I get an output such as `Tensor("grayscale_to_rgb_53:0", shape=(1, 3, 3), dtype=float32)` but the two digits after the underscore ("53" in this case) changes each run.

**Comment by mihaimaruseac** [comment]
We can print only the shape. See the doctest example in https://github.com/tensorflow/tensorflow/blob/f70c46f8cd2a91b390455827cda65b5b5fe92ef0/tensorflow/python/framework/ops.py#L446-L468

**Comment by WilliamHYZhang** [comment]
@mihaimaruseac thanks for the information, pushed the fix.

**Comment by mihaimaruseac** [comment]
One more thing: `>>>` should be followed by a space

**Comment by WilliamHYZhang** [comment]
Ok, I updated that in the PR. Thanks.

**Review by mihaimaruseac — COMMENTED** [review]
Last thing (apologies for not catching this in the previous review): Please remove `Usage Example` (as it is redundant) and remove one indentation level from the code blocks in the doctest.

**Comment by WilliamHYZhang** [comment]
No problem, pushed with the fix again. :)

**Review by mihaimaruseac — DISMISSED** [review]
Thank you. We'll take it from here

**Comment by gbaned** [comment]
@WilliamHYZhang Could you please address Ubuntu Sanity errors? Thanks!

**Comment by WilliamHYZhang** [comment]
Hm, that's odd. When I run the example in Google Colab (https://colab.research.google.com/drive/1S2g8mhOj-RZDmc3JvJW7FQqM-_ds4a6V), no error returns. What can I do to address these errors in Ubuntu Sanity?

**Comment by mihaimaruseac** [comment]
I don't understand why the log is not visible, let's try running the test again.

In the meanwhile, can you check with pylint if the code doesn't result in formatting errors?

**Comment by mihaimaruseac** [comment]
Probably you have to move the doctest before the `Args:` section, not after `Returns:`

**Comment by WilliamHYZhang** [comment]
Ok, pushed again. I moved the doctests like you suggested, thanks!

**Review by mihaimaruseac — DISMISSED** [review]

**Review by mihaimaruseac — COMMENTED** [review]
Instead of printing the shape (which is not giving enough information), can you print `converted.numpy()`? Here's my interactive session:

```console
>>> import tensorflow as tf
>>> original = tf.constant([[[1.0, 2.0, 3.0]]])    
>>> converted = tf.image.rgb_to_grayscale(original)
>>> converted
<tf.Tensor: shape=(1, 1, 1), dtype=float32, numpy=array([[[1.8149]]], dtype=float32)>
>>> converted.numpy()
array([[[1.8149]]], dtype=float32)
>>> converted.shape
TensorShape([1, 1, 1])
```

**Comment by WilliamHYZhang** [comment]
@mihaimaruseac thank you for the suggestion, I pushed the update.

**Review by mihaimaruseac — DISMISSED** [review]

**Comment by WilliamHYZhang** [comment]
Oh man, seems like the Ubuntu CPU test is failing again. Checked the status and seems like it returns ```[[[1.8149]]]``` instead of the expected ```array([[[1.8149]]], dtype=float32)```. Updated fix to reflect that. Also, it looks like it also was expecting the final ``` in the return as well, so I added a newline to hopefully fix that as well.

**Review by alextp — APPROVED** [review]

File: tensorflow/python/ops/image_ops_impl.py [hunk 3]
```diff
@@ -1866,7 +1866,15 @@ def rgb_to_grayscale(images, name=None):
   Outputs a tensor of the same `DType` and rank as `images`.  The size of the
   last dimension of the output is 1, containing the Grayscale value of the
   pixels.
+  
+  ```python
+  >>> original = tf.constant([[[1.0, 2.0, 3.0]]])
+  >>> converted = tf.image.rgb_to_grayscale(original)
+  >>> print(converted.numpy())
+  [[[1.8149]]]
```
Comment (alextp) [inline tensorflow/python/ops/image_ops_impl.py]: Maybe to 1.81... to avoid using too many digits of precision?

**Review by alextp — DISMISSED** [review]

**Comment by WilliamHYZhang** [comment]
CPU test failed again :(
Seems like it didn't return the commas in the np array for grayscale_to_rgb, just pushed to fix that. @alextp Is 4 decimal places of precision too much? Should I change that line to [[[1.81...]]], I don't know if that's syntactically correct. Thanks

**Comment by mihaimaruseac** [comment]
Yes, please convert to `[[[1.81...]` as `...` in doctest mean "don't try to match after this". 

Sorry it's taking so long and so many review rounds

**Review by mihaimaruseac — APPROVED** [review]

**Comment by WilliamHYZhang** [comment]
All tests now passing besides ```import/copybara — An error happened while migrating the change```, what does this mean?

**Comment by mihaimaruseac** [comment]
Let's try it again. I think it's an internal error.

**Comment by WilliamHYZhang** [comment]
Strange, the import/copybara error seems to be still happening.

**Comment by WilliamHYZhang** [comment]
@mihaimaruseac Shall I maybe refork the repo and create a new PR for this?

**Comment by mihaimaruseac** [comment]
No need to. There is an error on our side, there are people looking into it.

Sorry it's taking so long and sorry I couldn't respond earlier

**Pull request merged**